metals/src/main/scala/scala/meta/internal/metals/PackageProvider.scala

```diff
@@ -0,0 +1,48 @@
+package scala.meta.internal.metals
+
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.lsp4j.WorkspaceEdit
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+class PackageProvider(private val buildTargets: BuildTargets) {
+
+  def workspaceEdit(path: AbsolutePath): Option[WorkspaceEdit] = {
+    packageName(path).map(workspaceEdit(path, _))
+  }
+
+  private def packageName(path: AbsolutePath): Option[String] = {
+    // package.scala file needs to be handled separately
+    // as its package name is different than package name of normal scala file
+    if (path.isScala &&
+      path.toFile.length() == 0 &&
+      path.filename != "package.scala") {
+      val sourceItem = buildTargets.inverseSourceItem(path)
+      val relativeDirectory = sourceItem
+        .map(path.toRelative)
+        .flatMap(relativePath => Option(relativePath.toNIO.getParent))
+        .map(_.toString)
```

Comment:
maybe 
`map.(_.toString.replace("/", "."))`

then we don't need the next statement

Reply:
That didn't work on windows - changed.